demos/burgers-isotropic.py

```diff
@@ -0,0 +1,274 @@
+    return {"u": assemble(interpolate(as_vector([sin(pi * x), 0]), fs))}
+
+
+def get_qoi(mesh_seq, solutions, i):
+    R = FunctionSpace(mesh_seq[i], "R", 0)
+    dt = Function(R).assign(mesh_seq.time_partition[i].timestep)
+
+    def time_integrated_qoi(t):
+        u = solutions["u"]
+        return dt * inner(u, u) * ds(2)
+
+    return time_integrated_qoi
+
+
+n = 32
+meshes = [UnitSquareMesh(n, n, diagonal="left"), UnitSquareMesh(n, n, diagonal="left")]
+end_time = 0.5
+dt = 1 / n
+
+num_subintervals = len(meshes)
+time_partition = TimePartition(
+    end_time,
+    num_subintervals,
+    dt,
+    fields,
+    num_timesteps_per_export=2,
+)
+
+params = GoalOrientedMetricParameters(
+    {
+        "element_rtol": 0.001,
+        "maxiter": 35 if os.environ.get("GOALIE_REGRESSION_TEST") is None else 3,
+    }
+)
+
+mesh_seq = GoalOrientedMeshSeq(
+    time_partition,
+    meshes,
+    get_function_spaces=get_function_spaces,
+    get_initial_condition=get_initial_condition,
+    get_form=get_form,
+    get_solver=get_solver,
+    get_qoi=get_qoi,
+    qoi_type="time_integrated",
+    parameters=params,
+)
+
+# As in the previous adaptation demos, the most important part is the adaptor function.
+# The one used here takes a similar form, except that we need to handle multiple meshes
+# and metrics.
+# We then time integrate these to give the metric contribution from each subinterval.
+# Given that we use a simple implicit Euler method for time integration
+# in the PDE, we do the same here, too.
+#
+# Goalie provides functionality to normalise a list of metrics using *space-time*
+# normalisation. This ensures that the target complexity is attained on average across
+# all timesteps.
+#
+# Note that when adapting the mesh, we need to be careful to check whether convergence
+# has already been reached on any of the subintervals. If so, the adaptation step is
+# skipped. ::
+
+
+def adaptor(mesh_seq, solutions, indicators):
+    metrics = []
+    complexities = []
+
+    # Ramp the target average metric complexity per timestep
+    base, target, iteration = 400, 1000, mesh_seq.fp_iteration
+    mp = {
+        "dm_plex_metric": {
+            "target_complexity": ramp_complexity(base, target, iteration),
+            "p": 1.0,
+            "h_min": 1.0e-04,
+            "h_max": 1.0,
+        }
+    }
+
+    # Construct the metric on each subinterval
+    for i, mesh in enumerate(mesh_seq):
+        sols = solutions["u"]["forward"][i]
+        dt = mesh_seq.time_partition.timesteps[i]
+
+        # Define the Riemannian metric
+        P1_ten = TensorFunctionSpace(mesh, "CG", 1)
+
+        # At each timestep, recover metric of the solution
+        # vector. Then time integrate over the contributions
+        _metrics = []
+
+        for j, sol in enumerate(sols):
+
+            # get local indicator
+            indi = indicators["u"][i][j]
+            # local instance of Riemanian metric
+            _metric = RiemannianMetric(P1_ten)
```

Comment:
Similarly, `metric_j`?

Reply:
Good idea. Change implemented in 6dba2b8.